Makefile

```diff
@@ -69,6 +67,18 @@ COLORIZE=$(SED) ''/PASS/s//$(COLOR_PASS)/'' | $(SED) ''/FAIL/s//$(COLOR_FAIL)/''
 DOCKER_NAMESPACE?=jaegertracing
 DOCKER_TAG?=latest
 
+# list for multi-arch image publishing
+TARGET_ARCHS ?= amd64 arm64
+
+BUILDX_PUSH ?= false
+DOCKER_BUILD ?= docker build --build-arg=TARGETARCH=$(GOARCH)
```

Comment:
TARGETARCH is defined here but not in L76. How does it work, especially if the default value is removed from all Dockerfiles?

Reply:
Also, wouldn't this line be immediately evaluated with the current value of $(GOARCH) ?

Reply:
TARGETARCH is a platform args, which will be auto pass to Dockerfile when we use buildx.
https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

nope.  when we use `=` instead of `:=` in makefile, `$(GOARCH)` is as a ref value. will be evaluated  when `$(DOCKER_BUILD)` used.